frontend/src/app/workflows/workflows.component.ts

```diff
@@ -0,0 +1,166 @@
+  }
+
+  generate() {
+    this.privacyService.setStorage('workflows', this.content);
+
+    if (this.loading) return;
+
+    if (!this.content.includes('- workflow: ')) {
+      this.toastService.error('Lint Error', 'Needs at least one workflow note (best at the beginning)');
+      return;
+    }
+
+    // Replace tabs with two spaces for js-yaml and snakeyaml parser
+    this.content = this.content.replace(/\t/g, '  ');
+
+    const validYaml = this.lintService.lintYamlString(this.content, this.validate);
+
+    if (!validYaml) {
+      const errorMessage = this.lintService.evaluateErrorMessage(this.validate);
+      this.toastService.error('Lint Error', errorMessage);
+      return;
+    }
+
+    this.loading = true;
+
+    this.fulibWorkflowsService.generate(this.content).subscribe(
+      (answer) => {
+        this.generateResult = answer
+        this.loading = false;
+      },
+      (error: any) => {
+        this.loading = false;
+
+        let errorMessage = error.error.status + '\n';
+        errorMessage += error.error.message;
+
+        this.toastService.error('Parse Error', errorMessage);
+      }
+    );
+  }
+
+  @HostListener('window:message', ['$event'])
+  handleGlobalMessages(event: MessageEvent) {
+    const messageData = JSON.parse(event.data);
+    // It needs to be run in the NgZone because only then angular change detection gets a grip on the change
+    this.zone.run(() => {
+      this.newPageIndex = messageData.index;
+      this.currentDisplay = messageData.diagramType;
+    });
+
+    if (messageData.type === 'changeFrameWithToast') {
+      this.zone.run(() => {
+        this.toastService.success('Page Action', messageData.toastContent);
+      });
+    }
+  }
+
+  // Source: https://github.com/angular-split/angular-split/blob/main/src/app/examples/iframes/iframes.component.ts
+  dragStartHandler() {
+    this.showIframeHider = true
+  }
+
+  dragEndHandler() {
+    this.showIframeHider = false
+  }
+
+  splitGutterClick({gutterNum}: IOutputData) {
+    // By default, clicking the gutter without changing position does not trigger the 'dragEnd' event
+    // This can be fixed by manually notifying the component
+    // See issue: https://github.com/angular-split/angular-split/issues/186
+    this.split.notify('end', gutterNum)
+  }
+
+  openDocs() {
+    window.open('https://fujaba.github.io/fulibWorkflows/docs/definitions/', '_blank')
+  }
+
+  get url(): string {
+    if (!this.generateResult || !this.generateResult.board) {
+      return "";
+    }
+    return environment.workflowsUrl + '/workflows' + this.generateResult.board;
+  }
+
+  private setExample(exampleName: string) {
+    const url = `/assets/examples/workflows/${exampleName.replace(' ', '')}.es.yaml`;
+    this.http.get(url, {responseType: 'text'}).subscribe((value => {
+      if (value) {
+        this.content = value;
+        this.generate();
+      } else {
+        this.toastService.warn('Warning', 'Unknown Example.');
+      }
```

Comment:
Wouldn't the value always be truthy, but if someone really manages to select an invalid example, it would error with a 404?

Reply:
True. I put it into the error path of the observable